### ISSUE TITLE
Kev backend: Rust runtime with MLX on Apple Silicon, then adapter

### ISSUE THREAD

**Issue by codesoda** [codesoda/systemone#17]
Add a `kev` backend kind for [Kev](https://github.com/jaredpalmer/kev) (Apache-2.0): small, self-trainable Jev-like decision models. Each checkpoint is a rank-16 LoRA adapter plus a pointer head over a Qwen base. Kev's own server already speaks TypeSafe's System One API, so the SystemOne wire contract fits without translation.

Upstream facts (kev @ `557598f`, 2026-09-22):

- Current family: Kev-0.8B / 4B / 9B on Qwen3.5 bases (hybrid attention + Gated DeltaNet layers). Previous generation: Kev-0.6B / 4B / 8B on attention-only Qwen3 bases, still published (`jaredpalmer/kev-4b@qwen3`, `kev-8b`, `kev-0.6b`).
- Inference: encode `<state> … <q> … <opt> … </opt> … <decide>`, one isolated row per question on Qwen3.5 (the state prefix is computed once and its cache reused), and a pointer head that scores each `</opt>` hidden state against `<decide>`. A per-checkpoint temperature (about 2.1–2.4) is applied at load. `KEV_TEMPERATURE=1.0` gives raw logits.
- Apple Silicon: PyTorch MPS has no Gated DeltaNet kernels, so upstream runs the Qwen3.5 backbone through MLX (`kev/mlx_model.py`, mlx-lm). It merges the LoRA adapter in fp32 and keeps the encoder, pointer head and calibration unchanged. Kev-4B takes 721 ms per new state and 136 ms per repeated state on an M5. Parity with fp32 PyTorch is measured by `scripts/mlx_parity.py` / `tests/test_mlx.py`.
- Checkpoint layout: `adapter_config.json`, `adapter_model.safetensors`, tokenizer files and `head.pt`. `head.pt` is a PyTorch pickle holding head weights, temperature and metadata.
- `usage.output_tokens` counts tokens in the serialized answers, not generated tokens.

## Scope

- Add `kind = "kev"` alongside the existing kinds. Upstream stays an independent repository: a Rust runtime (e.g. `kev-rs`) owns the model, tokenizer, head and calibration. SystemOne owns the config, adapter and service, as with laya-rs and gliner2-rs.
- **MLX on macOS is required**, mirroring Laya. Add a portable `kev-cpu` feature and an Apple Silicon `kev-metal` feature (MLX via mlx-rs). `kev-metal` also enables the CPU backend for mixed configs, exactly like `laya-metal` / `laya-cpu`. `device = "cpu" | "metal"` selects the backend per instance. Asking for `metal` in a build without `kev-metal` is a clear configuration error, never a silent CPU fallback. Non-Apple builds must not pull in MLX.
- Qwen3.5 (Gated DeltaNet) support is the main runtime risk for the CPU path. Evaluate Candle and llama.cpp (already linked for openjev) for the hybrid backbone and record the chosen runtime and the rejected alternatives. If only the Qwen3 generation can reach parity on CPU at first, ship that honestly and document the limit. Do not claim Qwen3.5 CPU support.
- Load `head.pt` without executing pickle: use a restricted tensor-only reader or a converted, checksummed safetensors artifact produced from upstream. Pin model repos and revisions with SHA-256 checksums, as with the other local backends.
- Config sketch:

  ```toml
  [backends.local-kev]
  kind = "kev"
  enabled = false
  model = "kev-latest"

  [backends.local-kev.settings]
  run = "jaredpalmer/kev-4b"   # pinned revision
  device = "metal"             # cpu (kev-cpu build) or metal (kev-metal build)
  ```

## Acceptance gate

- Pinned Python reference outputs (goldens) for the selected checkpoints: fp32 PyTorch, plus upstream's MLX path on the same Apple machine. Record environment, precision, prefix-cache state and synchronization.
- The Rust runtime reaches numerical parity with upstream before integration. That covers tokenizer and special-token escaping, encoding, per-question row isolation, LoRA merge, pointer head and temperature. Tolerances must be justified against upstream's own MLX-vs-fp32 figures (e.g. Kev-4B max |dp| 0.025), and argmax flips must be reported. Packed questions and separately asked questions must match.
- The MLX path is built and exercised with real weights on Apple Silicon (`--features metal,laya-metal,gliner2,kev-metal` must build and serve together). The CPU path is tested in CI on Linux. Check that MLX (via laya), llama.cpp and ort still link in one binary with Kev enabled.
- The adapter maps `noul`, `choice` and `score` faithfully and does not renormalize. The `output_tokens` semantics are decided and documented, then added to the SDK smoke expectation table. Kev reports serialized-answer tokens, while the local backends report 0.
- The JS SDK smoke passes with `SYSTEMONE_SMOKE_BACKEND=kev` on both `cpu` and `metal`, using the same shared assertions as every other backend.
- Rust library and HTTP latency are benchmarked on the same workload as upstream's M5 numbers. No speedup is assumed.
- An upstream Python sidecar (`python -m kev.serve`) is a research baseline only and is never presented as the Rust implementation.
- README backend table, `docs/plans/cross-repo.md`, CHANGELOG and third-party licenses are updated. Qwen3.5 / Qwen3 base licenses and the Kev adapter weight licenses are confirmed before any release packaging.
